docs/common/istio-ambient-update.adoc

```diff
@@ -0,0 +1,784 @@
+
+In ambient mode, the IstioCNI component is responsible for traffic redirection. The component is compatible with multiple control plane versions during RevisionBased upgrades and continues to handle traffic redirection for both old and new control planes during the migration period.
+
+For detailed procedures on updating IstioCNI, see section "Updating IstioCNI".
+
+'''
+
+[[special-considerations-for-ambient-mode-upgrades]]
+== Special Considerations for Ambient Mode Upgrades
+
+[[ztunnel-daemonset-updates]]
+=== ZTunnel DaemonSet Updates
+
+The ZTunnel component runs as a DaemonSet on every node in the cluster. During upgrades:
+
+* **Rolling Updates:** ZTunnel uses a RollingUpdate strategy, updating one node at a time by default.
+* **Minimal Disruption:** While a node's ZTunnel pod is restarting, new connections may experience brief latency, but existing connections are maintained.
+* **Node-by-Node:** The update process ensures that at least one ZTunnel pod is always available on each node before proceeding to the next.
+* **Monitoring:** Monitor the ZTunnel DaemonSet rollout status:
+
+[source,bash]
+----
+$ kubectl rollout status daemonset/ztunnel -n ztunnel
+----
+
+[[istiocni-considerations]]
+=== IstioCNI Considerations
+
+The IstioCNI component is responsible for traffic redirection in ambient mode:
+
+* **Version Compatibility:** IstioCNI must be compatible with the ZTunnel version. Always upgrade IstioCNI before or alongside ZTunnel.
+* **DaemonSet Updates:** Like ZTunnel, IstioCNI runs as a DaemonSet and uses rolling updates.
+* **Traffic Redirection:** During the update, existing traffic redirection rules remain in place until the new version is applied.
+* **Verification:** After updating, verify IstioCNI pods are running on all nodes:
+
+[source,bash]
+----
+$ kubectl get pods -n istio-cni -o wide
+----
+
+[[control-plane-and-data-plane-version-skew]]
+=== Control Plane and Data Plane Version Skew
+
+In ambient mode, version skew between components is handled differently than in sidecar mode:
+
+* **Supported Skew:** The control plane (Istio) can typically be N+1 or N-1 relative to the data plane (ZTunnel).
+* **Testing Required:** Always test your specific version combinations in a non-production environment.
+* **Recommendation:** Keep all components (Istio, IstioCNI, ZTunnel) at the same version when possible.
+
+[[waypoint-proxy-compatibility]]
+=== Waypoint Proxy Compatibility
+
+If you have deployed waypoint proxies for L7 features:
+
+* **Automatic Updates:** Waypoint proxies automatically reference the active control plane revision.
+* **Gateway API:** Waypoint proxies are deployed using Kubernetes Gateway resources and will continue to function during upgrades.
+* **Recreation:** In rare cases, you may need to recreate waypoint Gateway resources if there are breaking changes between versions.
```

Comment:
Can you please share any reference for this statement.

Reply:
I think we can remove the "Recreation:" step.

Reply:
Removed under dbb70f5d